packages/back-end/src/util/organization.util.ts

```diff
@@ -68,10 +71,77 @@ export function roleToPermissionMap(
   return permissionsObj;
 }
 
-export function getUserPermissions(
+function mergePermissions(
+  existingPermissions: UserPermission,
+  existingRole: MemberRole | undefined,
+  teamInfo: TeamInterface | ProjectMemberRole,
+  org: OrganizationInterface
+) {
+  const newPermissions = roleToPermissionMap(teamInfo.role, org);
+  for (const newPermission in newPermissions) {
+    // If the user doesn't have permission, but the team role does, add it
+    if (
+      !existingPermissions.permissions[newPermission as Permission] &&
+      newPermissions[newPermission as Permission]
+    ) {
+      existingPermissions.permissions[newPermission as Permission] =
+        newPermissions[newPermission as Permission];
+    }
+  }
+
+  if (!existingRole) {
+    existingPermissions.limitAccessByEnvironment =
+      teamInfo.limitAccessByEnvironment;
+    existingPermissions.environments = teamInfo.environments;
+  } else {
+    const roles = getRoles(org);
+    const newRoleIndex = roles.findIndex((r) => r.id === teamInfo.role);
+    const existingRoleIndex = roles.findIndex((r) => r.id === existingRole);
+
+    const existingRoleIsEngineerOrExperimenter =
+      existingRole === "engineer" || existingRole === "experimenter";
+
+    const newRoleIsEngineerOrExperimenter =
+      teamInfo.role === "engineer" || teamInfo.role === "experimenter";
+
+    if (
+      existingRoleIsEngineerOrExperimenter &&
+      newRoleIsEngineerOrExperimenter
+    ) {
+      if (
+        // If limitAccessByEnvironment is the same for new and existing roles, we just concat the envs arrays
```

Comment:
So if we grant access to more environments via a team then we would honor that over the more restrictive environments list set globally. That makes sense

Reply:
Yup - this way, if globally they get access to "staging", and then they're also on  a team , which gives them access to "dev", they'll then be able to have access to both "dev", and "staging".